### DISCUSSION TITLE
Getting 4 vulnerabilities while running : npx hardhat init

### DISCUSSION THREAD

**Discussion by Rahul65911** [NomicFoundation/hardhat#5482] [Troubleshooting] [answered]
npm init --yes    
npm install --save-dev hardhat

Wrote to /home/rahul/Documents/Metaschool/test1/package.json:

{
  "name": "test1",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "description": ""
}

npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated glob@7.2.0: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported

added 267 packages, and audited 268 packages in 1m

52 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

npx hardhat init                                   
888    888                      888 888               888
888    888                      888 888               888
888    888                      888 888               888
8888888888  8888b.  888d888 .d88888 88888b.   8888b.  888888
888    888     "88b 888P"  d88" 888 888 "88b     "88b 888
888    888 .d888888 888    888  888 888  888 .d888888 888
888    888 888  888 888    Y88b 888 888  888 888  888 Y88b.
888    888 "Y888888 888     "Y88888 888  888 "Y888888  "Y888

Welcome to Hardhat v2.22.6

✔ What do you want to do? · Create a JavaScript project
✔ Hardhat project root: · /home/rahul/Documents/Metaschool/test1
✔ Do you want to add a .gitignore? (Y/n) · y
✔ Do you want to install this sample project's dependencies with npm (@nomicfoundation/hardhat-toolbox)? (Y/n) · y

# npm audit report

ws  7.0.0 - 7.5.9
Severity: high
ws affected by a DoS when handling a request with many HTTP headers - https://github.com/advisories/GHSA-3h5v-q93c-6h6q
fix available via `npm audit fix`
node_modules/@ethersproject/providers/node_modules/ws
  @ethersproject/providers  <=5.7.2
  Depends on vulnerable versions of ws
  node_modules/@ethersproject/providers
    ethers  5.0.0-beta.119 - 5.7.2
    Depends on vulnerable versions of @ethersproject/providers
    node_modules/eth-gas-reporter/node_modules/ethers
      eth-gas-reporter  >=0.2.26
      Depends on vulnerable versions of ethers
      node_modules/eth-gas-reporter

4 high severity vulnerabilities

To address all issues, run:
  npm audit fix

***
npm audit fix not solving the problem.
Is it node problem or other problem? Please enlighten me.

**Comment by cgewecke** [comment, marked as answer]
@Rahul65911 

As a workaround, these vulnerability warnings should be resolved if you upgrade to the latest version of `hardhat-gas-reporter` (2.2.0) with:

```
npm uninstall --save-dev hardhat-gas-reporter
npm install --save-dev hardhat-gas-reporter@latest
```

If that doesn't work would it be possible to [report the issue here][1]?

[1]: https://github.com/cgewecke/hardhat-gas-reporter/issues

**Reply by Rahul65911** [reply]
Thanks. It worked.

After reinstalling the gas-reporter I get 3 vulnerability that were solved using : `npm audit fix --force`
